### DISCUSSION TITLE
[Help Needed] Strategies to Speed Up `pprof -top` Execution for MOOSE Profiling Data

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#27467] [Q&A General] [answered]
Hello MOOSE Community,

I am using the MOOSE simulation framework for my computational research projects and have encountered an issue with performance analysis. Specifically, I've generated profiling data using the command:

```bash
MOOSE_PROFILE_BASE=run3_ mpiexec -n 32 ~/projocts/panda/panda-oprof -i s2_bicrystal_cp_elastic3.i > 01.log
```

This command successfully created 32 profiling files named `run3_[0-31].prof`. My challenge arises when trying to analyze these files with `pprof`. I use the following command to generate a performance report from one of the profiling files:

```bash
pprof -top ~/projocts/panda/panda-oprof run3_1.prof
```

However, the command execution is considerably slow and has not produced results even after running for more than 60 minutes.

I am looking for advice on how to efficiently analyze these profiling files. Are there any recommended practices within the MOOSE community for handling large profiling datasets or specific parameters that might accelerate the `pprof` tool's performance?

Thank you for your support and I look forward to any suggestions you might have.

Best regards,
Wei

**Comment by GiudGiud** [comment, marked as answer]
@lindsayad you had encountered massive slowdowns with pprof

I think the fix was

> unlinking libunwind and passing --enable-frame-pointers. I could have sworn that I had tried that before and gotten slow profiling

how involved is that?
Should we advocate using another profiler at this point?

**Reply by lindsayad** [reply]
What is your operating system and what is your binutils version? You can check this with `ld --version`. Did you read the note on the profiling page about issues with binutils version 2.37? You can read https://github.com/google/pprof/issues/680 and https://sourceware.org/pipermail/binutils/2022-February/119701.html. However, I will note that binutils version 2.40 no longer seems to have problems, so something must have been fixed between 2.37 and 2.40

**Reply by lindsayad** [reply]
So in summary, depending on your binutils version, downgrading or upgrading may fix your issue

**Reply by PengWei97** [reply]
Yes, your answer is correct. 

I solved this problem using `GNU ld (GNU Binutils) 2.29.1`. The version that needs a long time to wait is 2.38.

Thank you very much for your answer.
